### PR TITLE
Queues - Erica Case - Rideshare two

### PR DESCRIPTION
**Pull request by EricaJCasePhD**
# Ride Share
Congratulations! You're submitting your assignment!

## Comprehension Questions
| Question | Answer    |
| :------------- | :------------- |
| Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? |    I decided to do a mixin and then include rider, driver, and  trip as subclasses.  I made my decision because there were many repeated elements. |
| Describe a concept that you gained more clarity on as you worked on this assignment. | Inheritance--the thing I learned is that parent classes can access methods in subclasses.  |
| Describe a _nominal test_ that you wrote for this assignment. | Making sure that the find_all on a driver returned the correct information for the desired driver.   |
| Describe an _edge case test_ that you wrote for this assignment. |  I haven't written any yet.  |
| How do you feel you did in writing pseudocode first, then writing the tests and then the code? |  I didn't do this, which was undoubtedly a mistake. |

### CODE REVIEW

File: specs/driver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,105 @@
+require_relative 'spec_helper'
+
+describe "Driver" do
+  describe "initialize" do
+    let(:ok_driver) {Rideshare::Driver.new( { driver_id: 1, name: "OK Driver", vin: "1234567890123456B" }, :driver_id )}
+
+    let(:bad_vin_2short) {
```
Comment (droberts-sea) [inline specs/driver_spec.rb:7]: I like the names you've given these drivers - it's immediately obvious to the reader why one is good or bad.

**Comment by droberts-sea** [comment]
# Ride Share
## What We're Looking For

| Feature | Feedback    |
| :------------- | :------------- |
|  **Baseline** |   |
|   Used Git Regularly	|   This is an improvement, but I would still like to see more frequent commits and more descriptive messages. A message like "more stuff" does not meet the bar.	|
|   Answer comprehension questions	|   yes	|
|  **Driver** |   |
|   Uses the `all` method in the `find` method | |
|   Has appropriate edge-case tests for each method in the class |    |
|   Created a method that uses a method from the `Trip` object to retrieve the list of trips |  |
|   Created a method that uses the internal trips list to calculate the average rating | |
|  **Rider** |   |
|   Uses the `all` method in the `find` method | |
|   Has appropriate edge-case tests for each method in the class |    |
|   Created a method that uses a method from the `Trip` object to retrieve the list of trips |  |
|   Created a method that uses the internal trips method to retrieve the associated drivers | |
|  **Trip** |   |
|   Reads the CSV file in the `all` method | |
|   Has appropriate edge-case tests for each method in the class |    |
|   Created a method that uses a method from the `Driver` to retrieve the associated driver instance  |    |
|   Created a method that uses a method from the `Rider` to retrieve the associated rider instance |    |
|   Created a method to retrieve all trips by driver id |     |
|   Created a method to retrieve all trips by rider id |      |
|   **Overall**	|   	|

I'm going to let the table be - you know what's complete here and what isn't. To reiterate what we talked about in our 1-on-1, this is a good start, but attempting to add inheritance to the project was perhaps a little ambitious. Next time, make sure you've got the core requirements using things we've covered in class before you go above and beyond like this.
